Microsoft.DotNet.Interactive.Tests/LanguageServiceTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.Events;
+using Microsoft.DotNet.Interactive.LanguageService;
+using Microsoft.DotNet.Interactive.Tests.Utility;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Interactive.Tests
+{
+    public class LanguageServiceTests : LanguageKernelTestBase
+    {
+        public LanguageServiceTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        private Task<IKernelCommandResult> SendHoverRequest(KernelBase kernel, string code, int line, int character)
+        {
+            var command = new RequestHoverTextCommand(RequestHoverTextCommand.MakeDataUriFromContents(code), new Position(line, character));
+            return kernel.SendAsync(command);
+        }
+
+        [Theory]
+        [InlineData(Language.FSharp)]
+        [InlineData(Language.PowerShell)]
+        public async Task HoverOnUnsupportedLanguageServiceReturnsNothing(Language language)
+        {
+            using var kernel = CreateKernel(language);
+
+            var commandResult = await SendHoverRequest(kernel, "code", 0, 0);
+
+            commandResult
+                .KernelEvents
+                .ToSubscribedList()
+                .Should()
+                .ContainSingle<LanguageServiceNoResultProduced>();
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp, "var x = 1234;", 0, 10, "readonly struct System.Int32")]
+        public async Task HoverRequestReturnsExpectedResult(Language language, string code, int line, int character, string expected)
+        {
+            using var kernel = CreateKernel(language);
+
+            var commandResult = await SendHoverRequest(kernel, code, line, character);
+
+            commandResult
+                .KernelEvents
+                .ToSubscribedList()
+                .Should()
+                .ContainSingle<LanguageServiceHoverResponseProduced>()
+                .Which
+                .Contents
+                .Value
+                .Should()
+                .Contain(expected);
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp, "var x = 1; // hovering in a comment", 0, 20)]
```

Comment:
>0, 20 [](start = 76, length = 5)

For this kind of thing you can use MarkupTestFile, which is more readable and maintainable than specifying the ints separately.

Reply:
Thanks, I've updated most of the tests to do this.  There are some out-of-bounds tests that still must manually specify a line/character, though.